packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -971,11 +971,16 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 /// A [Navigator] observer that notifies [RouteAware]s of changes to the
 /// state of their [Route].
 ///
+/// ## Sample code
+///
 /// To make a [StatefulWidget] aware of its current [Route] state, implement
 /// [RouteAware] in its [State] and subscribe it to the [RouteObserver]:
 ///
 /// ```dart
 /// class RouteAwareWidgetState extends State<RouteAwareWidget> with RouteAware {
+///
+///   final RouteObserver routeObserver = new RouteObserver();
```

Comment:
As the documentation states, `RouteObserver` is a `NavigatorObserver`, so you'd most likely only instantiate one, register it with the `Navigator` and then register multiple Widgets (that want to be aware of changes to their current route) to the `RouteObserver`.